src/tools/web/kdb.js

```diff
@@ -0,0 +1,128 @@
+// kdb.js - small library to access Elektra's kdb via node.js
+const { exec } = require('child_process')
+
+// constants
+const ERR_KEY_NOT_FOUND = 'Did not find key'
+
+// KDBError
+function KDBError (message) {
+    this.name = 'KDBError'
+    this.message = message || ''
+}
+KDBError.prototype = Error.prototype
+
+// remove newline from the end of a string
+const trimNewline = (str) =>
+  str.substring(0, str.length - 1)
+
+// execute a script while catching and parsing errors
+const safeExec = (script) => new Promise((resolve, reject) =>
+  exec(script, (err, stdout, stderr) => {
+    if (err) {
+      const errors = err.message.split('\n')
+      // ignore error if it's "key not found"
+      if (!(errors.length > 1 && errors[1] === ERR_KEY_NOT_FOUND)) {
+        return reject(err)
+      }
+    }
+    if (stderr) {
+      if (stderr !== ERR_KEY_NOT_FOUND) {
+        return reject(new KDBError(stderr))
+      } else {
+        return resolve() // no key found, return no value
+      }
+    }
+    const result = trimNewline(stdout)
+    if (!result) {
+      return resolve() // empty result, return no value
+    }
+    return resolve(result)
+  })
+)
+
+// escape strings by surrounding them with ""
+const escapeValues = (template, ...values) =>
+  template.reduce((acc, part, i) => {
+    /*
+    Explanation of regular expression:
+    - $1 `(\\\\)*` Matches an even number of _backslashes_
+    - $4 Matches one _quote_ when there was an odd number of backslashes
+    - $5 Matches one _quote_ when there was an even number of backslashes
+
+    For instance in `\\\\\"`, $1 is `\\\\`, $4 is `"` and $5 is an empty string.
+    So `\\\\\"` gets replaced to itself.
+    In case of an even number of backslashes one backslash is added.
+
+    (source: @krit0n - https://github.com/ElektraInitiative/libelektra/pull/983#discussion_r83965059)
+    */
+    let val = values[i - 1].replace(/((\\\\)*)(\\(")|("))/g, '$1\\$4$5')
+    if (typeof val === 'string') val = `"${val}"`
```

Comment:
[This](https://github.com/ElektraInitiative/libelektra/pull/983#discussion_r83977227) is still an issue here.